### PR TITLE
C++ Streams support for hstring and IStringable 

### PR DESCRIPTION
**Pull request by Jaiganeshkumaran**
Implements #1195.

### CODE REVIEW

File: strings/base_stringable_format.h [hunk 1]
```diff
@@ -1,3 +1,8 @@
+inline std::wostream& operator<<(std::wostream& stream, winrt::Windows::Foundation::IStringable const& stringable)
```
Comment (sylveon) [inline strings/base_stringable_format.h:1]: Use a different file for this. base_stringable_format is for std::format. base_stringable_streams?

Comment (Jaiganeshkumaran) [reply strings/base_stringable_format.h]: Done.

File: strings/base_includes.h [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include <map>
 #include <memory>
 #include <optional>
+#include <ostream>
```
Comment (kennykerr) [inline strings/base_includes.h:13]: Is there no way to use a feature macro rather than including this in every single project? That seems problematic.

Comment (Jaiganeshkumaran) [reply strings/base_includes.h:13]: Maybe this could be included only if WINRT_LEAN_AND_MEAN isn't defined. Or should I define a separate macro?

Comment (kennykerr) [reply strings/base_includes.h:13]: Using `WINRT_LEAN_AND_MEAN` seems like a good idea. 👍

Comment (Jaiganeshkumaran) [reply strings/base_includes.h]: Done.

**Comment by kennykerr** [comment]
Looks good - just needs a basic test.

File: test/old_tests/UnitTests/streams.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        std::wstringstream ss;
+        winrt::hstring str = L"Hello\0World";
+        ss << str;
+        REQUIRE(ss.str() == str);
+    }
+
+    {
+        std::wstringstream ss;
+        winrt::Windows::Foundation::IStringable obj = winrt::make<stringable>();
+        ss << obj.ToString();
```
Comment (kennykerr) [inline test/old_tests/UnitTests/streams.cpp:33]: This tests `hstring` rather than `IStringable`.

**Review by kennykerr — APPROVED** [review]
Thanks!

**Pull request merged**